python/triton/language/core.py

```diff
@@ -1404,6 +1405,11 @@ def dot(input, other, acc=None, allow_tf32=True, max_num_imprecise_acc=None, out
     :param other: The second tensor to be multiplied.
     :type other: 2D tensor of scalar-type in {:code:`float16`, :code:`bfloat16`, :code:`float32`}
     """
+    if allow_tf32 is None:
+        if os.getenv("TRITON_F32_DEFAULT", "0") == "1":
```

Comment:
nit, in C++ we have a helper that accepts a variety of values meaning "true", we should probably accept the same values here.